### PR TITLE
Document absent optional subfield convention

### PR DESCRIPTION
**Pull request by PeytonT**
*Issue #, if available:* Somewhat related to https://github.com/amzn/ion-docs/issues/133 and https://github.com/amzn/ion-docs/issues/152.

*Description of changes:*

No functional change, proposed wording changes are only elaborating on existing behavior.
The specification calls for binary subfields to be elided whenever possible, but does so explicitly at each point where a subfield is discussed. Since this convention is required throughout, it can be noted as a general rule.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the Apache License 2.0.

### CODE REVIEW

**Review by tgregg — APPROVED** [review]
Makes sense to me. Will wait for one more approval.

File: docs/spec.md
Comment (zslayton) [inline docs/spec.md:138]: I was momentarily tripped up by the phrase 'without both', similar to the way one might wonder whether "or" meant "xor" in a given context. I'm fine with leaving it, but here's an alternate phrasing if you happen to agree:

```suggestion
* `0x51 0x80` and `0x52 0x80 0x00` are valid encodings of the decimal
value `0.` in Ion binary, but shall be encoded as `0x50` (with neither the 
`exponent` nor `coefficient` subfields) for hashing purposes. 
```

**Review by zslayton — APPROVED** [review]

**Pull request merged**
